public/components/mainMenu.js

```diff
@@ -0,0 +1,110 @@
+const { dialog } = require('electron');
+const updater = require('./updater');
+
+const openDialogOptions = {
+  buttonLabel: 'Open',
+  nameFieldLabel: 'Open:',
+  defaultPath: 'Protocol.netcanvas',
+  filters: [{ name: 'Network Canvas', extensions: ['netcanvas'] }],
+  properties: ['openFile'],
+};
+
+const openDialog = () =>
+  new Promise((resolve, reject) => {
+    dialog.showOpenDialog(openDialogOptions, (filename) => {
+      if (filename === undefined) { reject(); return; }
+      resolve(filename[0]);
+    });
+  });
+
+const openFile = window =>
+  () =>
+    openDialog()
+      .then(filePath => window.webContents.send('OPEN_FILE', filePath));
+
+const MenuTemplate = (window) => {
+  const menu = [
+    {
+      label: 'File',
```

Comment:
See notes about OPEN_FILE; this flow appears to be a no-op; we should probably remove the menu item for now. (Being able to open a protocol in NC would be great, though!)